hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/receiver/kafka/KafkaSingleThreadedMessageReceiver.java

```diff
@@ -160,6 +135,7 @@ public void stop() {
     @Override
     public void update(Subscription newSubscription) {
         this.subscription = newSubscription;
+        messageConverter.update(subscription);
```

Comment:
Right now we have the same subscription object in two places. Therefore, we have to synchronize its state, which is what you're doing in this line. 

In my opinion it would be better to remove the subscription field from the `KafkaConsumerRecordToMessageConverter`class and instead of this pass subscription as an argument of `convertToMessage` metchod

Reply:
Take a look that it is consistent with how it is done in `KafkaSingleThreadedMessageReceiver`. `subscription` could  be passed to `next()` and other methods in order to avoid mutability of it, but I guess someone picked this approach to avoid complex methods signatures. I'm not convinced which one option is better. Just wanted to let you know that it was a kind of tradeoff - I also don't like mutability. Let me know which one direction I should go with.